core/templates/dev/head/services/ExtensionTagAssemblerService.ts

```diff
@@ -17,22 +17,30 @@
  * the learner and editor views.
  */
 
-require('filters/string-utility-filters/camel-case-to-hyphens.filter.ts');
-require('services/HtmlEscaperService.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HtmlEscaperService } from 'services/HtmlEscaperService';
+import { CamelCaseToHyphensPipe } from
+  'filters/string-utility-filters/camel-case-to-hyphens.pipe';
 
 // Service for assembling extension tags (for interactions).
-angular.module('oppia').factory('ExtensionTagAssemblerService', [
-  '$filter', 'HtmlEscaperService', function($filter, HtmlEscaperService) {
-    return {
-      formatCustomizationArgAttrs: function(element, customizationArgSpecs) {
-        for (var caSpecName in customizationArgSpecs) {
-          var caSpecValue = customizationArgSpecs[caSpecName].value;
-          element.attr(
-            $filter('camelCaseToHyphens')(caSpecName) + '-with-value',
-            HtmlEscaperService.objToEscapedJson(caSpecValue));
-        }
-        return element;
-      }
-    };
+
+export class ExtensionTagAssemblerService {
+  constructor(private htmlEscaperService: HtmlEscaperService,
+              private camelCaseToHyphens: CamelCaseToHyphensPipe) {}
+
+  formatCustomizationArgAttrs(element: any,
+      customizationArgSpecs: Object): any {
```

Comment:
It's an element since we are using ```element.attr()``` in our code below but m not sure about the type...I tried a couple things like DomElement but didn't worked...so for now I'll add a todo for this

Reply:
sounds good.